racy condition in SegHead function

When I compile with GasNet I get a build error:
```
/src/ReductionMsg.chpl:964: error: Racy promotion of scalar method receiver
              cullOverReferences :   0.815 seconds
make: *** [Makefile:383: arkouda_server] Error 1
```

--- a/src/ReductionMsg.chpl
+++ b/src/ReductionMsg.chpl
@@ -960,9 +960,11 @@ module ReductionMsg
       const dom = makeDistDom(newSize);
       var ret = makeDistArray(dom, intype);
 
-      forall (oldSegStart, newSegStart, newLength) in zip(segments, newSegs, newSegLengths) with (const ref values, ref ret){
-        forall (r,v) in zip(ret[newSegStart..#newLength], values[oldSegStart..#newLength]) with (var agg = newSrcAggregator(intype)){
-          agg.copy(r, v);
+      forall (oldSegStart, newSegStart, newLength) in zip(segments, newSegs, newSegLengths) with (ref values,
+                                                                                            var agg = newDstAggregator(intype)){
+        var v = new lowLevelLocalizingSlice(values, oldSegStart..#newLength);
+        for i in 0..#newLength {
+          agg.copy(ret[newSegStart+i], v.ptr[i]);
         }
       }
       return ret;
@@ -980,9 +982,11 @@ module ReductionMsg
       const dom = makeDistDom(newSize);
       var ret = makeDistArray(dom, intype);
 
-      forall (oldSegEnd, newSegEnd, newLength) in zip(oldSegEnds, newSegEnds, newSegLengths) with (const ref values, ref ret){
-        forall (r,v) in zip(ret[(newSegEnd - newLength)..#newLength], values[(oldSegEnd - newLength)..#newLength]) with (var agg = newSrcAggregator(intype)){
-          agg.copy(r, v);
+      forall (oldSegEnd, newSegEnd, newLength) in zip(oldSegEnds, newSegEnds, newSegLengths) with (ref values,
+                                                                                            var agg = newDstAggregator(intype)){
+        var v = new lowLevelLocalizingSlice(values, (oldSegEnd - newLength)..#newLength);
+        for i in 0..#newLength {
+          agg.copy(ret[(newSegEnd - newLength)+i], v.ptr[i]);
         }
       }
       return ret;

--- a/src/ReductionMsg.chpl
+++ b/src/ReductionMsg.chpl
@@ -960,8 +960,10 @@ module ReductionMsg
       const dom = makeDistDom(newSize);
       var ret = makeDistArray(dom, intype);
 
-      forall (oldSegStart, newSegStart, newLength) in zip(segments, newSegs, newSegLengths) with (var agg = newSrcAggregator(intype)){
-          agg.copy(ret[newSegStart..#newLength], values[oldSegStart..#newLength]);
+      forall (oldSegStart, newSegStart, newLength) in zip(segments, newSegs, newSegLengths) with (const ref values, ref ret){
+        forall (r,v) in zip(ret[newSegStart..#newLength], values[oldSegStart..#newLength]) with (var agg = newSrcAggregator(intype)){
+          agg.copy(r, v);
+        }
       }
       return ret;
     }
@@ -978,8 +980,10 @@ module ReductionMsg
       const dom = makeDistDom(newSize);
       var ret = makeDistArray(dom, intype);
 
-      forall (oldSegEnd, newSegEnd, newLength) in zip(oldSegEnds, newSegEnds, newSegLengths) with (var agg = newSrcAggregator(intype)){
-          agg.copy(ret[(newSegEnd - newLength)..#newLength], values[(oldSegEnd - newLength)..#newLength]);
+      forall (oldSegEnd, newSegEnd, newLength) in zip(oldSegEnds, newSegEnds, newSegLengths) with (const ref values, ref ret){
+        forall (r,v) in zip(ret[(newSegEnd - newLength)..#newLength], values[(oldSegEnd - newLength)..#newLength]) with (var agg = newSrcAggregator(intype)){
+          agg.copy(r, v);
+        }
       }
       return ret;
     }

